components/camel-aws/camel-aws2-ses/src/main/java/org/apache/camel/component/aws2/ses/Ses2Endpoint.java

```diff
@@ -51,6 +56,15 @@ public void doStart() throws Exception {
         sesClient = configuration.getAmazonSESClient() != null
                 ? configuration.getAmazonSESClient()
                 : Ses2ClientFactory.getSesClient(configuration).getSesClient();
+
+        healthCheckRepository = HealthCheckHelper.getHealthCheckRepository(getCamelContext(),
+                ComponentsHealthCheckRepository.REPOSITORY_ID, ComponentsHealthCheckRepository.class);
+
+        if (healthCheckRepository != null) {
+            clientHealthCheck = new Ses2HealthCheck(this, getId());
+        }
+
+        healthCheckRepository.addHealthCheck(clientHealthCheck);
```

Comment:
@rhuan080 - I think we should be adding the health check within the if statement that checks if `healthCheckRepository` is null.

I fixed a similar case last week in https://github.com/apache/camel/pull/8994.

Reply:
Yes! I`ll send a PR to fix it.